src/connection.js

```diff
@@ -389,6 +366,12 @@ class Connection extends EventEmitter {
     };
   }
 
+  validateBooleanOptions(option, optionName) {
```

Comment:
How do you feel about making this a "regular" function instead of a method on `Connection`? Basically, just like `deprecateNonBooleanConfigValue` on `master`.